homeassistant/components/plugwise/switch.py

```diff
@@ -54,37 +53,25 @@ def __init__(
         self._attr_unique_id = f"{device_id}-{description.key}"
         self._attr_name = coordinator.data.devices[device_id].get("name")
 
+    @plugwise_command
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the device on."""
-        try:
-            state_on = await self.coordinator.api.set_switch_state(
```

Comment:
What does the return value here mean? Is that a response from the api? If we get an actual response from the api it could be wise to apply it immediately since it's not just an assumption.

Reply:
It's a response that the requests succeeded

https://github.com/plugwise/python-plugwise/blob/96f2963ba7b25cc0abcdd21e90e320c3f567b6c5/plugwise/smile.py#L628-L629

However, it raises if it fails and the coordinator refreshes the state directly afterward. So, it doesn't add anything to handle it here.

Reply:
Basically: the module returns`True` except when a switch is in locked state or an exception in the request to the local API.

Reply:
There is even a case in the library where this can return `False`, if it detects the switch is already in the requested state. So, the existing logic was actually bugged.

<https://github.com/plugwise/python-plugwise/blob/main/plugwise/smile.py#L623-L624>

Reply:
Ok.